lib/netext/grpcext/conn.go

```diff
@@ -58,9 +61,22 @@ type clientConnCloser interface {
 
 // Conn is a gRPC client connection.
 type Conn struct {
-	raw clientConnCloser
+	addr   string
+	shares atomic.Uint64
+	raw    clientConnCloser
 }
 
+var (
+	// Is there any other way?
```

Comment:
it seems like yes

it seems like we need a new struct like `connectionPool` (a better name). This struct could be initialized and live in the `RootModule`
https://github.com/grafana/xk6-grpc/blob/main/grpc/grpc.go#L15-L26 This struct could also take the big part of the management of the connections probably

And pass it to the client via the NewClient
https://github.com/grafana/xk6-grpc/blob/main/grpc/grpc.go#L59-L63